src/main/java/de/tum/in/www1/artemis/service/util/structureoraclegenerator/JavaClassDiff.java

```diff
@@ -276,6 +295,52 @@ private boolean methodNamesAreEqual(JavaMethod solutionMethod, JavaMethod templa
         return solutionMethod.getName().equals(templateMethod.getName());
     }
 
+    private boolean typesAreEqual(JavaClass solutionClass, JavaClass templateClass) {
+        return solutionClass.getName().equals(templateClass.getName());
+    }
+
+    private boolean annotationsAreEqual(List<JavaAnnotation> solutionAnnotations, List<JavaAnnotation> templateAnnotations) {
```

Comment:
True, although I think we should still keep the check in two different methods, so that the code is easier to understand. What should definitely be refactored is the following: 
```
    private boolean basicListEquality(List l1, List l2) {
        if (l1 == null && l2 == null) {
            return true;
        }
        if (l1 == null || l2 == null) {
            return false;
        }
        if (l1.isEmpty() && l2.isEmpty()) {
            return true;
        }
        
        return l1.size() == l2.size();
    }
```

Reply:
I would propose to improve this in another PR ;-)